src/appleseed/renderer/kernel/lighting/pathtracer.h

```diff
@@ -684,6 +714,11 @@ bool PathTracer<PathVisitor, VolumeVisitor, Adjoint>::process_bounce(
             sample);
 
         next_ray.m_max_roughness = m_clamp_roughness ? sample.m_max_roughness : 0.0f;
+        if (sample.m_mode == ScatteringMode::Diffuse && !vertex.m_albedo_aov_saved)
+        {
+            vertex.m_albedo = sample.m_value.m_albedo;
```

Comment:
What do you think of introducing PathVisitor::on_first_diffuse_bounce(const PathVertex& vertex) and the visitor can then copy the albedo from the vertex to ShadingComponents?
We could undo the changes in PathTracer::trace

Reply:
I tried to add a method to PathVisitor but then all the lighting engines required to add it to their implementation of PathVisitor. So I had to add dummy methods there. Is that better than modifying PathTracer?
I'll try to add it and add a commit.